storage/src/cache/cachedfile.rs

```diff
@@ -394,27 +407,44 @@ impl FileCacheEntry {
             let meta = self
                 .get_blob_meta_info()?
                 .ok_or_else(|| einval!("failed to get blob meta object"))?;
-            let zran_index = meta.get_zran_index(chunks[0].id());
-            let (ctx, _) = meta
-                .get_zran_context(zran_index)
-                .ok_or_else(|| einval!("failed to get ZRan context for chunk"))?;
+            let zran_index = meta.get_zran_index(chunks[0].id())?;
+            let (ctx, _) = meta.get_zran_context(zran_index)?;
             let blob_start = ctx.in_offset;
-            let zran_index = meta.get_zran_index(chunks[chunks.len() - 1].id());
-            let (ctx, _) = meta
-                .get_zran_context(zran_index)
-                .ok_or_else(|| einval!("failed to get ZRan context for chunk"))?;
+            let zran_index = meta.get_zran_index(chunks[chunks.len() - 1].id())?;
+            let (ctx, _) = meta.get_zran_context(zran_index)?;
             let blob_end = ctx.in_offset + ctx.in_len as u64;
             (blob_start, blob_end)
         } else if self.is_batch {
-            let meta = self
-                .get_blob_meta_info()?
-                .ok_or_else(|| einval!("failed to get blob meta object"))?;
+            let first_chunk = &chunks[0];
+            let last_chunk = &chunks[chunks.len() - 1];
+
+            // Get blob meta info iff the chunk is batch chunk.
+            let mut meta: Option<Arc<BlobCompressionContextInfo>> = None;
```

Comment:
How about ?
```
match (first_chunk.is_batch(), last_chunk.is_batch()) {
   (true, true) => ...
   (true, false) => ...
   (false, true) =>...
   (false, false) => ...
}
```

Reply:
A helper function as below should simplify the implementation.

Reply:
Helper `BlobCCI` is added for simplification.